### DISCUSSION TITLE
generic table

### DISCUSSION THREAD

**Discussion by fastlink30** [dotnetcore/BootstrapBlazor#2604] [Q&A] [answered]
Hi,
i want create new component with table inside bind to generic class,
on the project attached i have example of what i do (see table demo page and component/generictable)
not understand where is the error when i run the app..
[BootstrapBlazorApp.OnlyServer1.zip](https://github.com/dotnetcore/BootstrapBlazor/files/13692871/BootstrapBlazorApp.OnlyServer1.zip)

**Comment by ArgoZhang** [comment, marked as answer]
@fastlink30 

### 1 add constrains

```
public partial class GenericTable<TValue> where TValue : class, new()
{
    [Parameter] public IEnumerable<TValue> data { get; set; }
}
```

### 2 declare TValue

```
<BootstrapBlazorApp.OnlyServer1.Component.GenericTable TValue="GenericClass" data="genericData" />
<BootstrapBlazorApp.OnlyServer1.Component.GenericTable TValue="GenericClass2" data="genericData2" />
```
